spot_ros2_control/launch/spot_ros2_control.launch.py

```diff
@@ -14,11 +17,78 @@
 
 from spot_driver.launch.spot_launch_helpers import get_login_parameters, spot_has_arm
 
+THIS_PACKAGE = "spot_ros2_control"
+
+
+def create_controllers_config(spot_name: str, has_arm: bool) -> None:
+    """Writes a configuration file used to put the ros2 control nodes into a namespace.
+    This is necessary as if your ros2 control nodes are launched in a namespace, the configuration yaml used
+    must also reflect this same namespace when defining parameters of your controllers.
+
+    The generated file will be put in `<package share of spot_ros2 control>/config/spot_default_controllers.yaml`.
+
+    Args:
+        spot_name (str): Name of spot. If it's the empty string, the default controller file with no namespace is used.
+        has_arm (bool): Whether or not your robot has an arm. Necessary for defining the joints that the forward
+                        position controller should use.
+    """
+
+    arm_text = "with_arm" if has_arm else "without_arm"
+    template_filename = os.path.join(
+        get_package_share_directory(THIS_PACKAGE), "config", f"spot_default_controllers_{arm_text}.yaml"
+    )
+    output_filename = os.path.join(get_package_share_directory(THIS_PACKAGE), "config", "spot_default_controllers.yaml")
+
+    with open(template_filename, "r") as template_file:
+        config = yaml.safe_load(template_file)
+
+        if spot_name:
+            forward_position_controller_joints = config["forward_position_controller"]["ros__parameters"]["joints"]
+            config["forward_position_controller"]["ros__parameters"]["joints"] = [
+                f"{spot_name}/{joint}" for joint in forward_position_controller_joints
```

Comment:
@khughes-bdai meta: not for now, because I know the controller we are using is a stock controller, but this would be so much simpler if the controller understood that a `namespace` (its own namespace?) should be applied to the given joints.